modulus/metrics/climate/healpix_loss.py

```diff
@@ -189,3 +189,68 @@ def forward(self, prediction, target, average_channels=True):
             return th.sum(ocean_mean_err) / self.lsm_sum
         else:
             return ocean_mean_err / self.lsm_var_sum
+
+
+class WeightedOceanMSE( th.nn.MSELoss ):
+    """
+    Ocean MSE class offers impementaion for MSE loss with: 
+    1) weighted by a land-sea-mask field. 
+    2) weighted by channel (e.g. sic more than sst)
+    """
+    def __init__(
+        self,
+        lsm_file: str,
+        open_dict: dict = {
+            'engine':'zarr'},
+        selection_dict: dict = {
+            'channel_c':'lsm'},
+        weights: Sequence = [],
+    ):
+        """
+        """
+        super().__init__()
+        self.device = None
+        self.lsm_file = lsm_file
+        self.lsm_ds = None
+        self.open_dict = open_dict
+        self.selection_dict = selection_dict
+        self.lsm_tensor = None 
+        self.lsm_sum_calculated = False
+        self.lsm_sum = None
+        self.lsm_var_sum = None
+        self.loss_weights = th.tensor(weights)
+
+    def setup(self, trainer):
+        """
+        reshape lsm and put on device 
+        pushes weights to cuda device 
+        """
+        ### 1. OCEAN PREP ###
+        self.lsm_ds = xr.open_dataset(self.lsm_file,**self.open_dict).constants.sel(self.selection_dict)
+        # 1-lsm gives the percentage of pixel that has ocean
```

Comment:
I am not sure I know the discussion leading to this, but you are using "percentage of pixel that has ocean" implying a weight that is not just 0 or 1 but could be in between. What informs this choice ? for the model resolution a data point is either full ocean or full atmosphere right ?

Reply:
The ERA5 lsm is 0s and 1s. When we interpolate to HPX we do not force the interpolated values to be strictly binary. This is because we want coastal cells (which include both ocean and land) to truthfully represent the fraction of land/ocean they contain.

Reply:
There wasn't much discussion regarding how best to treat coastal pixels. The simplest solution-- using the the HEALPix mapped version of LSM (with values between 0 and 1)-- worked well so we kept it. I suggest keeping it simple unless someone can think of a compelling reason to change.